yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtil.scala

```diff
@@ -223,4 +225,43 @@ object YarnSparkHadoopUtil {
     )
   }
 
+  /**
+   * Expand environment variable using Yarn API.
+   * If environment.$$() is implemented, return the result of it.
+   * Otherwise, return the result of environment.$()
+   * Note: $$() is added in Hadoop 2.4.
+   */
+  def expandEnvironment(environment: Environment): String = {
```

Comment:
I don't know, the method name feels a little bit long. I'd be ok with just `expand` (since the argument already tells you what's being expanded), but not a big deal.